Core/NoOpRunner.Core/Shapes/Player.cs

```diff
@@ -214,15 +239,77 @@ public void Update(MessageDto message)
             if (message.MessageType != MessageType.PlayerUpdate) 
                 return;
             
+            Console.WriteLine("Observer: Player, say Hello World");
+            
             var messageDto = message.Payload as PlayerStateDto;
 
             State = messageDto.State;
             IsLookingLeft = messageDto.IsLookingLeft;
 
             CenterPosX = messageDto.CenterPosX;
             CenterPosY = messageDto.CenterPosY;
+            
+        }
+
+        public void TakePowerUp(PowerUps powerUp)
+        {
+            PlayerOnePowerUps.TakePowerUp(powerUp);
+        }
+        public void Display(Canvas canvas)
```

Comment:
This is ugly. We shouldn't strongly couple player data with its rendering. Is there really no better way to do this?

Reply:
Yes